cloud/src/meta-service/meta_service_txn.cpp

```diff
@@ -2640,10 +2750,43 @@ void MetaServiceImpl::check_txn_conflict(::google::protobuf::RpcController* cont
                         running_table_ids.end(), result.begin());
                 result.resize(iter - result.begin());
                 if (result.size() > 0) {
-                    response->set_finished(false);
-                    LOG(INFO) << "skip timeout txn count: " << skip_timeout_txn_cnt
-                              << " total iteration count: " << total_iteration_cnt;
-                    return;
+                    finished = false;
+                    std::vector<std::tuple<std::variant<int64_t, std::string>, int, int>> out;
+                    std::string_view key_view = k;
+                    key_view.remove_prefix(1);
+                    int ret = decode_key(&key_view, &out);
+                    if (ret != 0) [[unlikely]] {
+                        // decode version key error means this is something wrong,
+                        // we can not continue this txn
+                        LOG(WARNING) << "failed to decode key, ret=" << ret << " key=" << hex(k);
+                    } else {
+                        DCHECK(out.size() == 5) << " key=" << hex(k) << " " << out.size();
+                        const std::string& decode_instance_id = std::get<1>(std::get<0>(out[1]));
+                        int64_t db_id = std::get<0>(std::get<0>(out[3]));
+                        int64_t txn_id = std::get<0>(std::get<0>(out[4]));
+                        std::string conflict_txn_info_key =
+                                txn_info_key({decode_instance_id, db_id, txn_id});
```

Comment:
make this part a separate function which should be called `get_txn_info_key_from_txn_running_key()`